### PR TITLE
[docs] Add BugSnag guide

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

This PR:
- Adds BugSnag guide.
- Removes previously added redirect.

# Test Plan

Run docs locally and visit: http://localhost:3002/guides/using-bugsnag/ or see preview.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by brentvatne — APPROVED** [review]

File: docs/pages/guides/using-bugsnag.mdx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+title: Use BugSnag
+description: A guide on installing and configuring BugSnag for end-to-end error reporting and analytics.
+hideTOC: true
+---
+
+import { BoxLink } from '~/ui/components/BoxLink';
+import { BookOpen02Icon } from '@expo/styleguide-icons';
+
+[BugSnag](https://www.bugsnag.com) is a stability monitoring solution that provides rich, end-to-end error reporting and analytics to reproduce and fix errors with speed and precision. BugSnag supports the full stack with open-source libraries for more than[50 platforms](https://www.bugsnag.com/platforms), including [React Native](https://docs.bugsnag.com/platforms/react-native/react-native/).
```
Comment (keith-kurak) [inline docs/pages/guides/using-bugsnag.mdx:10]:
```suggestion
[BugSnag](https://www.bugsnag.com) is a stability monitoring solution that provides rich, end-to-end error reporting and analytics to reproduce and fix errors with speed and precision. BugSnag supports the full stack with open-source libraries for more than [50 platforms](https://www.bugsnag.com/platforms), including [React Native](https://docs.bugsnag.com/platforms/react-native/react-native/).
```
missing space

File: docs/pages/guides/using-bugsnag.mdx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+With BugSnag, developers and engineering organizations can:
+
+- **Stabilize:** Innovate faster by knowing when to build new features versus fix bugs. Use the release health dashboard, stability scores and targets, and built-in alerts via email, Slack, PagerDuty, and more.
+- **Prioritize:** Improve customer experience by identifying and prioritizing bugs that have the greatest impact on app stability. Analyze issues grouped by root cause and sorted by business impact, customer segmentation, A/B testing and experiment results.
+- **Fix:** Increase productivity by spending less time on reproducing and fixing bugs. Utilize powerful diagnostic data, full stacktraces and automatic breadcrumbs.
+
+## Integration
+
+See the integration guide below for instructions on how to add BugSnag to your Expo apps to report JavaScript errors. You can also read the [blog post](https://www.bugsnag.com/blog/build-apps-in-expo-with-bugsnag) announcing BugSnag's Expo integration.
```
Comment (keith-kurak) [inline docs/pages/guides/using-bugsnag.mdx:20]:
```suggestion
See the integration guide below for instructions on how to add BugSnag to your Expo apps to report JavaScript errors. This guide includes instructions for uploading source maps for updates published with [EAS Update](/eas-update/introduction/).
```
The blog link is dead, so suggested removing that.

That they do include info about EAS Update support seems relevant to users arriving here to see what bug reporting options they have. It's included in the below link, but if we wanted to link to that specifically, it's here: https://docs.bugsnag.com/platforms/react-native/expo/#eas-update

**Review by keith-kurak — APPROVED** [review]
Left a small spacing fix and a suggestion for calling out EAS Update support

File: docs/pages/guides/using-bugsnag.mdx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+With BugSnag, developers and engineering organizations can:
+
+- **Stabilize:** Innovate faster by knowing when to build new features versus fix bugs. Use the release health dashboard, stability scores and targets, and built-in alerts via email, Slack, PagerDuty, and more.
+- **Prioritize:** Improve customer experience by identifying and prioritizing bugs that have the greatest impact on app stability. Analyze issues grouped by root cause and sorted by business impact, customer segmentation, A/B testing and experiment results.
+- **Fix:** Increase productivity by spending less time on reproducing and fixing bugs. Utilize powerful diagnostic data, full stacktraces and automatic breadcrumbs.
+
+## Integration
+
+See the integration guide below for instructions on how to add BugSnag to your Expo apps to report JavaScript errors. This guide includes instructions for uploading source maps for updates published with [EAS Update](/eas-update/introduction/).
```
Comment (amandeepmittal) [inline docs/pages/guides/using-bugsnag.mdx:20]:
```suggestion
See the integration guide below for instructions on adding BugSnag to your Expo apps to report JavaScript errors. It also includes instructions for uploading source maps for updates published with [EAS Update](/eas-update/introduction/).
```

**Pull request merged**